### PR TITLE
Ensure root UserInputManager is alive

### PR DESCRIPTION
**Pull request by nyquillerium**
Currently, the very top level of the draw hierarchy never becomes alive. We need to make sure UserInputManager is always alive because checks dependent on parent life will always fail as a result.

### CODE REVIEW

**Comment by smoogipoo** [comment]
No test? I'd also override `ShouldBeAlive => true` just for sanity

**Comment by nyquillerium** [comment]
Tests coming in the other PR. Just pring both for now for visibility since this pr doesn't really have an explanation without the other.

**Comment by peppy** [comment]
Please make constructor `internal protected` and add a comment why `IsAlive` needs to be set.

File: osu.Framework.Tests/Platform/UserInputManagerTest.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            AddAssert("UserInputManager is alive", () =>
+            {
+                using (var client = new TestHeadlessGameHost(@"client", true))
+                {
+                    return client.CurrentRoot.IsAlive;
+                }
+            });
+        }
+
+        private class TestHeadlessGameHost : HeadlessGameHost
```
Comment (peppy) [inline osu.Framework.Tests/Platform/UserInputManagerTest.cs]: i dunno, this doesn't make much sense.

at this point you might as well just do

```csharp
Asset.IsTrue(new UserInputManager().IsAlive);
```

**Comment by peppy** [comment]
At which point it's not really a visual test, as i said in my comment.

If we're testing this it may make more sense to test gameHost.Root.IsAlive?

**Comment by nyquillerium** [comment]
That was what the previous form of the test did, though upon reflection it may make sense to make a normal gamehost instead of a headless one because it needs to be able to run the game in order to have a root. Would no longer need to manually create the root like I did there.

**Comment by peppy** [comment]
You can still run a headless game host, for what it's worth.

**Pull request merged**